docs/tutorials/handler.md

```diff
@@ -0,0 +1,389 @@
+first step in configuring credentials for publishing is to log in to
+Atomist using the `rug repositories login` command.  You will be
+prompted for your GitHub user credentials.
+
+```console
+$ rug repositories login
+
+
+The Rug CLI needs your GitHub login to identify you.
+
+The command will create a GitHub Personal Access Token with scope 'read:org'
+which you can revoke any time on https://github.com/settings/tokens.  Your
+password will not be displayed or stored. Your sensitive information will not
+be sent to Atomist; only to api.github.com.
+
+  → Username : .....
+  → Password : .....
+
+  Please provide a MFA code
+  → MFA code : .....
+
+Successfully logged in to GitHub and stored token in ~/.atomist/cli.yml
+```
+
+!!! important ""
+    No passwords are stored, only a unique personal access token with
+    read:org scope.
+
+!!! note "Two-Factor Authentication"
+    If your GitHub account is configured to use two-factor
+    authentication, you will be prompted for the MFA code as shown
+    above.
+
+Once you are logged in, you can run the `rug repositories configure`
+to save your publishing configuration to your Rug CLI configuration
+file.
+
+```console
+$ rug repositories configure
+
+→ Repositories
+  <unique identifier of your team's repository>
+    <your unique team's repository url here>
+```
+
+You're now all set to publish your handlers project.
+
+### Publishing
+
+You use the Rug CLI `publish` command to publish a Rug project archive
+to Atomist.  You execute the `rug publish` command from the handlers
+project directory:
+
+```console
+$ rug publish
+Resolving dependencies for yourorg:handlers:0.1.0 ← local completed
+Invoking TypeScript Compiler on .ts files
+  Created .atomist/handlers/CloseIssueThanksHandler.js.map
+  Created .atomist/handlers/CloseIssueThanksHandler.js
+Processing script sources completed
+Loading yourorg:handlers:0.1.0 ← local into runtime completed
+  Created META-INF/maven/yourorg/handlers/pom.xml
+  Created .atomist/manifest.yml
+  Created .atomist/metadata.json
+Generating archive metadata completed
+  Uploading yourorg/handlers/0.1.0/handlers-0.1.0.zip → slackteam (186 kb) succeeded
+  Uploading yourorg/handlers/0.1.0/handlers-0.1.0.pom → slackteam (927 bytes) succeeded
+  Uploading yourorg/handlers/0.1.0/handlers-0.1.0-metadata.json → slackteam (762 bytes) succeeded
+  Downloading yourorg/handlers/maven-metadata.xml ← slackteam (382 bytes) succeeded
+  Uploading yourorg/handlers/maven-metadata.xml → slackteam (334 bytes) succeeded
+Publishing archive into remote repository completed
+
+→ Archive
+  ~/workspace/handlers/.atomist/target/handlers-0.1.0.zip (186 kb in 165 files)
+
+→ URL
+  https://atomist.jfrog.io/atomist/SLACKTEAM/yourorg/handlers/0.1.0/handlers-0.1.0.zip
+
+Successfully published archive for yourorg:handlers:0.1.0
+```
+
+Your new handler will now be triggered when issue events occur in your
+Atomist environment.  To test that everything has published correctly
```

Comment:
I'm not sure `events occur in your atomist environment.` Perhaps `events occur in your environment that flow to Atomist` (as earlier in the document), or simply `events occur in your environment`.